specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/StackHCI/edgeMachineJob.tsp

```diff
@@ -0,0 +1,741 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+using Azure.Core;
+
+@armProviderNamespace("Microsoft.AzureStackHCI")
+namespace Microsoft.AzureStackHCI;
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" ""
+@added(Versions.v2025_12_01_preview)
+@doc("Cluster Jobs resource")
+@parentResource(EdgeMachine)
+model EdgeMachineJob
+  is Azure.ResourceManager.ProxyResource<EdgeMachineJobProperties> {
+  ...ResourceNameParameter<
+    Resource = EdgeMachineJob,
+    KeyName = "jobsName",
+    SegmentName = "jobs",
+    NamePattern = "^[a-zA-Z0-9-_]{3,63}$"
+  >;
+}
+
+@added(Versions.v2025_12_01_preview)
+@armResourceOperations
+interface EdgeMachineJobs {
+  @doc("Get a EdgeMachineJob")
+  get is ArmResourceRead<EdgeMachineJob>;
+
+  @doc("Create a EdgeMachineJob")
+  createOrUpdate is ArmResourceCreateOrUpdateAsync<EdgeMachineJob>;
+
+  @doc("Delete a EdgeMachineJob")
+  delete is ArmResourceDeleteWithoutOkAsync<EdgeMachineJob>;
+
+  @doc("List EdgeMachineJob resources by EdgeMachines")
+  list is ArmResourceListByParent<EdgeMachineJob>;
+}
+
+@@doc(EdgeMachineJob.name, "Name of EdgeMachineJob");
+
+//////////////////// models ////////////////////
+
+@added(Versions.v2025_12_01_preview)
+@doc("EdgeMachine Job properties")
+@discriminator("jobType")
+model EdgeMachineJobProperties {
+  @doc("Job Type to support polymorphic resource.")
+  @visibility(Lifecycle.Create, Lifecycle.Read)
+  jobType: EdgeMachineJobType;
+
+  @doc("Deployment mode to trigger job.")
+  deploymentMode?: DeploymentMode;
+
+  @doc("Job provisioning state")
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState;
+
+  @doc("Unique, immutable job id.")
+  @visibility(Lifecycle.Read)
+  jobId?: string;
+
+  @doc("The UTC date and time at which the job started.")
+  @visibility(Lifecycle.Read)
+  startTimeUtc?: utcDateTime;
+
+  @doc("The UTC date and time at which the job completed.")
+  @visibility(Lifecycle.Read)
+  endTimeUtc?: utcDateTime;
+
+  @doc("Status of Edge device job.")
+  @visibility(Lifecycle.Read)
+  status?: JobStatus;
+
+  @doc("error details.")
+  @visibility(Lifecycle.Read)
+  error?: Azure.ResourceManager.Foundations.ErrorDetail;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Job Type supported.")
+union EdgeMachineJobType {
+  string,
+
+  @doc("Job to collect logs from the device.")
+  CollectLog: "CollectLog",
+
+  @doc("Job to provide remote support to the device.")
+  RemoteSupport: "RemoteSupport",
+
+  @doc("Job to provision operating system in the device.")
+  ProvisionOs: "ProvisionOs",
```

Comment:
>Os [](http://example.com/codeflow?start=25&length=2)

nit - Only when the acronym is three characters or more, PascalCase is used. Here I would suggest `ProvisionOS`